### ISSUE TITLE
Manual Import: Unable to load manual import items

### ISSUE THREAD

**Issue by egyptianbman** [lidarr/Lidarr#2035]
**Describe the bug**
I see there are a couple of other issues that are closed about this, it seems not enough information was provided to track down the problem. It happens very consistently for me so I'll have a go at it.

When there are too many files for the manual import call to process, my browser is receiving a 504. This happens at exactly one minute after the request to `/api/v1/manualimport` is made.

**To Reproduce**
1. Go to `/wanted/missing`
2. Click on `Manual Import`
3. Select a directory with enough files that it'll take longer than one minute for the `ImportDecisionMaker` to read the files and `IdentificationService` to identify the albums.

**Expected behavior**
I expect the api call to eventually succeed, even if takes a few minutes.

**Screenshots**
Loading breakdown for the api call:
![image](https://user-images.githubusercontent.com/323600/109460905-fd2f1800-7a26-11eb-8841-6367b6a7c096.png)

![image](https://user-images.githubusercontent.com/323600/109460936-0ddf8e00-7a27-11eb-8cbd-3ca508212d9f.png)

**Platform Information (please complete the following information):**
 - OS: Arch
 - Docker: Yes
 - Mono or .NET Version (System -> Status): .NET Core 3.1.10
 - Browser and Version (Only needed for UI issues): Brave
 - Lidarr Version: 0.8.0.2075
 - Lidarr Branch: default/unchanged

**Trace Logs**
Trace logs don't give anything useful. I checked the docker container to see if it's running a proxy and found:
```
root@c9411d8d94b0:/# ps auxf
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root        6341  0.6  0.0  20192  3936 pts/0    Ss   00:48   0:00 bash
root        6354  0.0  0.0  36088  3312 pts/0    R+   00:48   0:00  \_ ps auxf
root           1  0.0  0.0    204     4 ?        Ss   Feb28   0:00 s6-svscan -t0 /var/run/s6/services
root          35  0.0  0.0    204     4 ?        S    Feb28   0:00 s6-supervise s6-fdholderd
root         252  0.0  0.0    204     4 ?        S    Feb28   0:00 s6-supervise lidarr
abc          255  1.9  0.7 4844636 519788 ?      Ssl  Feb28  14:26  \_ /app/lidarr/bin/Lidarr -nobrowser -data=/config
```
So, it looks like lidarr is running directly without a proxy.

The log in the first proxy (this proxies directly to the container) shows:
````
GET /api/v1/manualimport?artistId=0&folder=%2Fpath%2Fto%2Ffiles&filterExistingFiles=true&replaceExistingFiles=false HTTP/1.1" 499 0 "https://domain.com/wanted/missing" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.192 Safari/537.36
````

The log in the second proxy (this proxies to the first proxy) shows:
```
GET /api/v1/manualimport?artistId=0&folder=%2Fpath%2Fto%2Ffiles&filterExistingFiles=true&replaceExistingFiles=false HTTP/2.0" 504 569 "https://domain.com/wanted/missing" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.192 Safari/537.36
```

It looks like the first proxy sends a 499 which the second proxy passes on as a 504. Please let me know if you need any additional information; thanks!

**Comment by ta264** [comment]
One of the proxies has a timeout set. You'll need to remove it.

**Comment by egyptianbman** [comment]
Thanks for the quick response, @ta264! My setup seems to be more complex than I thought. It turns out I have a third proxy that I needed to update! If anyone else runs into this issue, the following are required exactly as follows in nginx:

```nginx
proxy_http_version 1.1;
proxy_set_header Connection "";
proxy_set_header Host               $host;
proxy_set_header X-Real-IP          $remote_addr;
proxy_set_header X-Forwarded-For    $proxy_add_x_forwarded_for;
keepalive_timeout 21600;
```
Furthermore, it seems Cloudflare won't proxy requests that don't respond in under 100 seconds. There are workaround that send back 8kb of data to trigger Cloudflare to wait i.e. https://github.com/marcialpaulg/Fixing-Cloudflare-Error-524/blob/master/long-running-proccess-FIXED.php but for now, I've disabled the cloudflare proxy.
